File: ProxyPattern/ProxyPattern/Proxy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace ProxyPattern
{
    interface Picture
    {
        Image getImage();
    }


    class Proxy : Picture
    {
        String file;
        MyImage img;
        public Proxy(String file)
        {
            this.file = file;
        }
        public Image getImage()
        {
            if (img == null)
            {
                img = new MyImage(file);
            }
            return img.getImage();
        }
    }


    class MyImage : Picture
    {
        Image img;
        public MyImage(String fileName)
        {
            img = Image.FromFile(fileName);
        }
        public Image getImage()
        {
            return img;
        }
    }
}

File: Proxy/src/Proxy.java
import javax.imageio.ImageIO;
import java.awt.*;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;

class MyImage implements Picture{
    BufferedImage img;

    public MyImage(String fileName){
        File file = new File(fileName);
        try {
            img = ImageIO.read(file);
        } catch (IOException exception) {
            exception.printStackTrace();
       }
    }

    public Image getImage(){
        return img;
    }
}

public class Proxy implements Picture{
    MyImage img;
    String file;
    public  Proxy(String file){
        this.file = file;
    }
    @Override
    public Image getImage() {
        if(img == null){
            img = new MyImage(file);
        }
        return img.getImage();
    }
}

